### PR TITLE
[DX-1655 TT-12355] added docs on gateway log format

### PR DESCRIPTION
**Pull request by LLe27**
### For internal users - Please add a Jira DX PR ticket to the subject!
[TT-12355](https://tyktech.atlassian.net/browse/TT-12355)
[DX-1655](https://tyktech.atlassian.net/browse/DX-1655)

---

### Preview Link

[Preview Link](https://deploy-preview-5346--tyk-docs.netlify.app/docs/nightly/log-data/#setting-log-format-only-available-for-the-gateway)

### Description

Added documentation and examples on the new feature that allows the control of the log format.

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR

[TT-12355]: https://tyktech.atlassian.net/browse/TT-12355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DX-1655]: https://tyktech.atlassian.net/browse/DX-1655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by LLe27** [comment]
@dcs3spp not sure what the reasoning is but in our main docs page this page in particular doesn't even show up in the search which is a bit unfortunate since it is quite a key feature. Please let me know how we can improve the search of this page. Thanks

File: tyk-docs/content/log-data.md [hunk 1]
```diff
@@ -54,6 +54,22 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
 
 If unset or left empty, it will default to `info`. 
 
+### Setting log format (only available for the Gateway)
+
+As of Tyk Gateway version `v5.6.0`, you can control log format using the `TYK_GW_LOGFORMAT` environment variable. By default, logs are in standard format, but setting `TYK_GW_LOGFORMAT` to `json` will output logs in JSON format.
```
Comment (titpetric) [inline tyk-docs/content/log-data.md]: This needs adjustment to list the 5.3 LTS release as well

Comment (titpetric) [reply tyk-docs/content/log-data.md]: Document as above, TYK_LOGFORMAt / TYK_LOGLEVEL (`_GW` prefix isn't in use in this doc)

Comment (LLe27) [reply tyk-docs/content/log-data.md]: Gotcha I wasn't sure cause I only saw `v5.6.0` listed in the fixed versions. Just to confirm both LTS `v5.3` and `v5.6.x` will include these changes?

Comment (LLe27) [reply tyk-docs/content/log-data.md]: Will update the configs accordingly -- I guess I wanted to make it clear that it's only available with the Gateway but this makes sense.

File: tyk-docs/content/log-data.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 date: 2017-03-24T12:53:50Z
 title: Logging system and API events
-tags: ["logging", "observability", "system events", "logs"]
+tags: ["logging", "observability", "system events", "logs", "json", "log format"]
```
Comment (dcs3spp) [inline tyk-docs/content/log-data.md]: @LLe27 The tags are used for search indexing and headings. Maybe also add tags for JSON, logging, logging format. In a separate comment I have made suggestions for sub headings for Standard and JSON format logging

File: tyk-docs/content/log-data.md [hunk 3]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
 
 If unset or left empty, it will default to `info`. 
 
+### Setting log format (only available for the Gateway)
+
+As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in standard format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
+
+**Standard Format:**
```
Comment (dcs3spp) [inline tyk-docs/content/log-data.md]:
```suggestion
##### Standard logging format
```

File: tyk-docs/content/log-data.md [hunk 4]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
...
+```
+time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
+```
+
+**JSON Format:**
```
Comment (dcs3spp) [inline tyk-docs/content/log-data.md]:
```suggestion
##### JSON logging format
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks @LLe27 I have made suggestions relating to additional search tags and made suggestions to add section headings for the examples. When config PR has been synched into docs should then be able to add a link to the new environment variable in gateway-config.md

File: tyk-docs/content/log-data.md (hunk 2 above)
Comment (LLe27) [reply tyk-docs/content/log-data.md]: Gotcha thanks! Will remove json for now but I think log format for search indexing would be worth keeping.

**Review by dcs3spp — APPROVED** [review]
Thanks @LLe27 I have approved changes and once link has been added to env var when config synced into docs we can merge and release. @titpetric if the changes requested from @LLe27 are ok would you be able to peer approve?

File: tyk-docs/content/log-data.md [hunk 5]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
...
+##### Standard logging format
+```
+time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
+```
+
+##### JSON logging format
+```
+{"level":"info","msg":"Tyk API Gateway v5.6.0","prefix":"main","time":"2024-09-05T09:01:23-04:00"}
+```
+
```
Comment (titpetric) [inline tyk-docs/content/log-data.md]:
```suggestion
As a general performance tuning note, the `json` output provides less allocation overhead than the standard logger. When tuning for performance, it's recommended to configure the JSON logging format.
```

File: tyk-docs/content/log-data.md (hunk 1 above)
Comment (titpetric) [reply tyk-docs/content/log-data.md]: @LLe27 for now, how about opening a dashboard PR? :)

Comment (LLe27) [reply tyk-docs/content/log-data.md]: @titpetric that's on my todo list, with your guidance of course! ;)

File: tyk-docs/content/log-data.md [hunk 6]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
 
 If unset or left empty, it will default to `info`. 
 
+### Setting log format (only available for the Gateway)
+
+As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in standard format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
```
Comment (sharadregoti) [inline tyk-docs/content/log-data.md]:
```suggestion
As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
```
This environment variable accepts only two options: `default` or `json`. Let's refer the standard format as the default format?

Reference Code: https://github.com/TykTechnologies/tyk/blob/master/log/log.go#L71

File: tyk-docs/content/log-data.md [hunk 7]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
...
+
+##### Standard logging format
```
Comment (sharadregoti) [inline tyk-docs/content/log-data.md]:
```suggestion
##### Default logging format
```

File: tyk-docs/content/log-data.md [hunk 8]
```diff
@@ -54,6 +54,21 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
...
+```
+time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
+```
+
+##### JSON logging format
+```
+{"level":"info","msg":"Tyk API Gateway v5.6.0","prefix":"main","time":"2024-09-05T09:01:23-04:00"}
+```
+As a general performance tuning note, the `json` output provides less allocation overhead than the standard logger. When tuning for performance, it's recommended to configure the JSON logging format.
```
Comment (sharadregoti) [inline tyk-docs/content/log-data.md]:
```suggestion
{{< note >}}
**Note**  
As a general performance tip, the `json` output format incurs less memory allocation overhead than the default logger. For optimal performance, it's recommended to configure logging in the JSON format.
{{< /note >}}
```

@LLe27 When you are referring to less allocation, it is with respect to `memory` right?

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: tyk-docs/content/log-data.md (hunk 6 above)
Comment (LLe27) [reply tyk-docs/content/log-data.md]: @sharadregoti correct! Thanks for the review.

File: tyk-docs/content/log-data.md (hunk 8 above)
Comment (LLe27) [reply tyk-docs/content/log-data.md]: @sharadregoti correct!

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/log-data.md [hunk 9]
```diff
@@ -54,6 +54,24 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
 
 If unset or left empty, it will default to `info`. 
 
+### Setting log format (only available for the Gateway)
+
+As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
```
Comment (sharadregoti) [inline tyk-docs/content/log-data.md]:
```suggestion
As of Tyk Gateway versions LTS `v5.3.6` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
```

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: tyk-docs/content/log-data.md (hunk 9 above)
Comment (LLe27) [reply tyk-docs/content/log-data.md]: I think it's slotted for `v5.6.0` now. @titpetric can you confirm which version of `v5.3.x` it is merged to? Thanks

Comment (letzya) [reply tyk-docs/content/log-data.md]: we'll put 5.3.6 based on the changelog of 5.3.6 FYI @andyo-tyk @lghiur

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/log-data.md [hunk 10]
```diff
@@ -54,6 +54,24 @@ Sometimes you will want to have more detailed logging for the Tyk Gateway than f
...
+
+As of Tyk Gateway versions LTS `v5.3.6` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
+
+##### Default logging format
+```
+time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
+```
+
+##### JSON logging format
+```
```
Comment (letzya) [inline tyk-docs/content/log-data.md:67]:
```suggestion
```json
```

**Pull request merged**